Add cancel button to loading screen

deveee opened the pull request.

deveee committed:
Add cancel button to loading screen

deveee committed:
Use real event receiver function from inputhandler.

Also move the button position to center below progress bar.

MoNTE48 commented on src/client/game.cpp:
```diff
@@ -1618,6 +1712,22 @@ bool Game::getServerContent(bool *aborted)
+	GUIButton *cancel_button = GUIButton::addButton(guienv, rect, texture_src, nullptr, ID_cancelButton, text);
+	cancel_button->setStyles(StyleSpec::getButtonStyle());
+	cancel_button->setAlignment(gui::EGUIA_LOWERRIGHT, gui::EGUIA_LOWERRIGHT, gui::EGUIA_LOWERRIGHT, gui::EGUIA_LOWERRIGHT);
+
```
No `delete[] text;` here?

deveee committed:
Allow to cancel loading after media received

deveee committed:
Merge branch 'main' into cancel-button

MoNTE48 commented:
![image](https://github.com/MultiCraft/MultiCraft/assets/6764600/053e4d96-0ebd-409c-875d-266ddcfe3081)

Swiping on a button (without actually pressing it) on Android breaks the touch interface.

deveee committed:
No need to send translated mouse event to inputhandler.

Fixes hidden touchscreen gui after game start.

grorp committed:
Fix all cached media being loaded at once on the main thread

deveee committed:
Don't create the button on single player loading screen

luk3yx committed:
Try and scale cancel button like formspecs

deveee committed:
Merge branch 'main' into cancel-button

deveee committed:
Make cancel button working while loading media.

Also store a pointer to connect aborted variable in client to avoid passing it
to each function in mediaclient, nodedef etc.

SkyBuilder1717 commented:
its actually so useful, why dont you make it till the end?
